src/Domain/ValetToken/constants.ts

```diff
@@ -0,0 +1,8 @@
+// conformant to https://day.js.org/docs/en/display/format#list-of-all-available-formats
+export const dateFormat = 'YYYY-MM-DDTHH:mm:ss[Z]'
+// 2 hours
+export const defaultExpiresAfterSeconds = 2*60*60
+// 5 minutes
+export const minExpiresAfterSeconds = 5*60
+// 7 days
+export const maxExpiresAfterSeconds = 7*24*60*60
```

Comment:
all of those consts should be part of some class for example:
```
class Foo {
  static readonly DEFAULT_EXPIRATION_TIME = 7200
  private readonly  DATE_FORMAT = '...'
  //...
}
```

Reply:
Let me preface this by saying that I'm fine with all caps and constants in classes. I'll adjust to whatever is the preferred coding style. Consistency is paramount.

***

Now a brief rationale why this might be fine. Less complexity. A module does the job of organizing constants just fine. No need for any class-related infrastructure to be related to this. And answer questions like which class do I put it in and does it really belong there or how do a name the class that I'll create...

BTW I haven't followed the tradition of naming constants with all caps out of habit and I have the habit because it's unnecessary in modern languages, especially with static typing and a good editor (in the purely hypothetical case that you need to be aware that an identifier refers to a constant you can just hover over it).

***

To be clear and dispel any doubts I gotta say here that I always try to argue only in the cases where I feel it's sensible and may lead to some improvement or if not that, then at least an interesting discussion.

Once we agree on some principle and it's sensible enough I'll follow it unless over time the sensibility turns out to have been misjudged. 

If my style of conversation might sound robotic (to use a phrase found in SN guidelines) sometimes, please note that this is only my habitual manner of thinking and writing on certain topics.

Anyway I might have had too much coffee today.

Reply:
the all-caps notation is not a requirement. Was just an example on my side. What I want more is to avoid throw-all-in-there piles like infamous "Utils" or "Constants" - I think almost in all cases constants are in the context of a given class - whether it's a model or a service or some other concept.  Given the fact that we want everything to follow OOP in all the server side services it would fit best to put constants into respective classes. Wdyt?

Reply:
Understood. I agree that throwing things into a big bag of everything is not the way to go and everything should be sensibly organized if possible.

So if I come up with or you suggest a more appropriate place for these constants, I'll put them there. :)

***

# Discussion

I think the existence of patterns like `Utils` or `Constants` is justified. True that we can easily go wrong with that if we only keep adding to these without first trying to fit a value into a more appropriate place and without refactoring to extract values into better places (if any).

But sometimes there are no obvious or sensible places, especially for highly shared or generic functions and values. In such case it might be sensible to put the thing in an independent staging area and put it elsewhere only when it becomes clear that it belongs there. This might be better than introducing wrong or unnecessary abstractions or state to satisfy a paradigm.

In this case (`Domain/ValetToken/constants.ts`) the bag is small and contextual, so I would say it's ok. We could also have a `ValetTokenConstants` or a (static?) `ValetToken` class with static/readonly/immutable fields to the same effect. But a module can already be viewed as a simpler version of that. Once it becomes necessary to create a more specialized class or put the variables in an existing class, it can be done.

Of course we can have a strict rule forbidding `utils` or `constants` to avoid the bad of this pattern. However it does not guarantee that what we do instead will be better.

Personally I like to try to apply everything in moderation, including programming paradigms (and so I like the flexibility of TypeScript or Scala better than Java which tries to shoehorn everything into it's version of OOP or Haskell which tries to shoehorn everything into ~math~ FP) -- what's more important is keeping things simple.

Reply:
To be honest I've seend too many `utils` and `constatns` go wrong so I don't use that anti-pattern at all and have always found a sensible place where to put constants that make sense in the domain logic. 

As a rule of thumb:
- think if this consts is used only for purposes of a given class/method -> if so put it in that class as `private readonly ...`
- If it's used in broader scope think of the Domain Context it touched. So for example `minExpiresAfterSeconds` would be the domain context of ValetToken and I'd put that in that model

Reply:
I've seen both bad and sensible instances of these, as with most patterns. Anyway no constants and utils it is. I'll try to apply the rule. Should a model imply a class or can a module be treated as a model?

Reply:
> Should a model imply a class or can a module be treated as a model?

Can you give an example?

Reply:
> > Should a model imply a class or can a module be treated as a model?
> 
> Can you give an example?

With a module the simplest way for me to satisfy conceptual OOP here would be to e.g. move the constants to the `ValetToken` module and treat that module as a model.

Alternatively I could put a `ValetToken` class in there with the constants as its `public static readonly` fields.

Both ways provide the same encapsulation and allow qualified names (although with a module it's optional and you can import a constant directly). A module is simpler.

I am assuming that "model" here is DDD terminology (not super-well versed in that).